Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut

```diff
@@ -182,27 +184,41 @@ void function SpawnHardpoints()
 		// spawnpoints are CHardPoint entities
 		// init the hardpoint ent
 		int hardpointID = 0
-		if ( spawnpoint.kv.hardpointGroup == "B" )
+		string group
+		if((GetMapName()=="mp_homestead")&&(!spawnpoint.HasKey("hardpointGroup")))
```

Comment:
Seeing these checks out of nowhere is kinda confusing, you should add a `ShouldHotfixHardpoint(hardpoint)` function (or something like that) that does the map and key check. Adding a comment that explains the hotfix to that function also helps.

Reply:
I encapsulated the get hardpoint group in a function. now the check is only done there and in the rest of the code we can just call `GetHardpointGroup(hardpoint)`